frontends/analytics/src/components/charts/RagQueries.tsx

```diff
@@ -84,40 +152,53 @@ export const RagQueries = (props: RagQueriesProps) => {
         </div>
       }
     >
-      <div>
-        <Show
-          fallback={<div class="py-8 text-center">Loading...</div>}
-          when={ragQueriesQuery.data}
-        >
-          {(data) => (
-            <Table
-              fallback={<div class="py-8 text-center">No Data</div>}
-              headerClass="px-2"
-              class="my-4"
-              headers={
-                <Tr>
-                  <Th>Message</Th>
-                  <Th>RAG Type</Th>
-                  <Th>Results</Th>
-                </Tr>
-              }
-              data={data()}
-            >
-              {(row) => (
-                <Tr>
-                  <Td fullWidth={true}>{row.user_message}</Td>
-                  <Td class="min-w-[100px]">
-                    {ALL_FAKE_RAG_OPTIONS.find(
-                      (rag) => rag.value === row.rag_type,
-                    )?.label || row.rag_type}
-                  </Td>
-                  <Td class="text-center">{row.results.length}</Td>
-                </Tr>
-              )}
-            </Table>
-          )}
-        </Show>
-      </div>
+      <Show
+        fallback={<div class="py-8 text-center">Loading...</div>}
+        when={ragQueriesQuery.data}
+      >
+        {(data) => {
+          const table = createSolidTable({
+            get data() {
+              return data();
+            },
+            state: {
+              pagination: {
+                pageIndex: pages.page(),
+                pageSize: 10,
+              },
+            },
+            columns: columns(),
+            getCoreRowModel: getCoreRowModel(),
+            manualPagination: true,
+          });
+          const usage = queryClient.getQueryData<RAGUsageResponse>([
+            "rag-usage",
+            { filter: props },
+          ]);
```

Comment:
Feels like this should just be in the main body of the component function. We have so much logic in the rendering on the other sites and its super messy. Would like to avoid that on analytics lol

Reply:
Thats super fair, I tried that first and stuff just wouldn't update, let me give it another try